<a href="https://colab.research.google.com/github/Altsybulski/Python-for-DA/blob/main/hw_6_2_%D0%A1%D1%82%D0%B0%D1%82%D0%B8%D1%81%D1%82%D0%B8%D1%87%D0%BD%D1%96_%D1%82%D0%B5%D1%81%D1%82%D0%B8%20Tsybulski.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Домашнє завдання 6: Статистичні тести

0. Зчитайте дані з `data.csv` та запишіть значення колонки data у форматі np.ndarray в змінну data. Ми будемо далі працювати з цією змінною.

In [ ]:
import pandas as pd
import numpy as np
import scipy.stats as stats
from statsmodels.stats.weightstats import ztest

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
df = pd.read_csv('drive/MyDrive/Data Analysis 7.0/data/data_statistics/data_1.csv')

In [ ]:
data_1 = np.array(df['data'])

Запустіть код нижче. Для коректної робити всіх подальших методів дані мають бути у вигляді одновимірного вектора, тобто мати розмір (100,).

In [ ]:
data_1.shape

(100,)

1. Зчитані дані - це сума покупок в доларах наших покупців на сайті протягом липня 2022 року.

До цього місяця, ми мали продажі в середньому на чек $20. Чи вийшло у нас статистично значущо **підвищити** середній чек за останній місяць?

Оберіть статистичний тест (z-test чи t-test). Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чеку.

**Підказка**: тут тест буде одновибірковий.

Представимо 2 гіпотези:

Нульова гіпотеза (H₀): Середнє значення чека за липень 2022 року не відрізняється від 20 доларів (середнє значення не змінилося).


In [ ]:
# Очікуване середнє значення (середній чек до липня 2022 року)
expected_mean = 20
# Рівен значущості
alpha = 0.05

In [ ]:
# Виконання t-тесту для однієї вибірки за допомогою scipy
t_stat, p_value = stats.ttest_1samp(data_1, expected_mean)
# Виводимо результати
print(f"t-статистика: {t_stat}")
print(f"p-значення: {p_value}")

t-статистика: -0.9953477389335053
p-значение: 0.3219937839011988


In [ ]:
# Висновок
if p_value < alpha:
    print("Ми маємо статистично значущу різницю в середньому чеку (відкидаємо H₀).")
else:
    print("Немає статистично значущої різниці в середньому чеку (не відкидаємо H₀).")

Немає статистично значущої різниці в середньому чеку (не відкидаємо H₀).


2. Пройшов час, і ми зібрали дані про продажі за серпень 2022 року, які зберігли у файл `data_2.csv`. Продажів було дещо більше, але чи змінився стат. значущо середній чек порівняно з липнем 2022? Перевірте це використовуючи ztest.

  Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чека між двома місяцями.

  **Підказка**: треба провести двовибірковий ztest з Н0 - що середні значення в липні та серпні 2022 року - однакові. Для кращого розуміння ситуації, можливо вам буде цікаво переглянути, які середні показники чеку були в кожен з місяців. Порівняйте свій висновок з простого спостереження середніх значень і висновок статистичного тесту. Чи вони збігаються?

In [ ]:
df = pd.read_csv('drive/MyDrive/Data Analysis 7.0/data/data_statistics/data_2.csv')

In [ ]:
data_2 = np.array(df['data'])

In [ ]:
data_2.shape

(120,)

In [ ]:
# Порівняємо просто середні значення для кожного місяця data_1(Липень), data_2(Серпень):
print('mean(Липень) = %.2f, mean(Серпень) = %.2f' % (np.mean(data_1), np.mean(data_2)))

mean(Липень) = 19.38, mean(Серпень) = 19.37


Висновок: Немає статистично значущої різниці в середньому чеку між двома місяцями - не відкидаємо H₀.

In [ ]:
# Виконуємо тест.
ztest(data_1, data_2, value=0)

(0.005246673456808504, 0.9958137794593356)

Статистика для двох вибірок z-критерію становить 0.0052, а відповідне p-value дорівнює 0.9958.

In [ ]:
# Висновок
if p_value < alpha:
    print("Ми маємо статистично значущу різницю в середньому чеку (відкидаємо H₀).")
else:
    print("Немає статистично значущої різниці в середньому чеку (не відкидаємо H₀).")

Немає статистично значущої різниці в середньому чеку (не відкидаємо H₀).


Оскільки p-value > 0.05, у нас немає достатньо доказів, щоб відкинути нульову гіпотезу.
Іншими словами, у нас немає значущої різниці у розмірі середнього чека між двома місяцями 2022 року, середні значення в липні та серпні 2022 року - однакові.